client/components/social-buttons/facebook.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import SocialLogo from 'social-logos';
+
+export default class FacebookLoginButton extends React.Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			FB: null,
+		};
+		this.handleClick = this.handleClick.bind( this );
+	}
+
+	componentDidMount() {
+		this.loadDependency().then( FB => {
+			FB.init( {
+				appId: this.props.appId,
+				xfbml: this.props.xfbml,
+				cookie: this.props.cookie,
+				version: this.props.version,
+			} );
+			this.setState( { FB } );
+		} );
+	}
+
+	loadDependency() {
+		return new Promise( resolve => {
+			if ( window.FB ) {
+				resolve( window.FB );
+			} else {
+				( function ( d, s, id ) {
+					const element = d.getElementsByTagName( s )[0];
+					const fjs = element;
+					let js = element;
+					if ( d.getElementById( id ) ) { return; }
+					js = d.createElement( s ); js.id = id;
+					js.src = '//connect.facebook.net/en_US/sdk.js';
+					fjs.parentNode.insertBefore( js, fjs );
+				}( document, 'script', 'facebook-jssdk') );
+
+				window.fbAsyncInit = () => {
+					resolve( window.FB );
+					delete window.fbAsyncInit;
+				};
+			}
+		} );
+	}
+
+	handleClick() {
+		this.state.FB.login( response => {
+			if ( this.props.responseHandler ) {
+				this.props.responseHandler( response );
+			}
+		}, { scope: this.props.scope } );
+	};
+
+	render() {
+		// Show an empty button if Facebook Connect API is not loaded
+		if ( ! this.state.FB ) {
```

Comment:
Super minor, but instead of a presence check for a reference to `window.FB`, what do you think about setting a boolean like `isLoaded`? This might be a little clearer, and is the minimum state needed to represent the render states here (as there are only two).